scripts/metadata_comparison/metadata_comparison/comparer.py

```diff
@@ -4,101 +4,288 @@
-                        help='GCS base path to the directory containing JSONs produced by digester')
-    parser.add_argument('--output-gcs-file-path', metavar='OUTPUTGCSFILE', type=gcs_path_regex_validator, nargs=1,
-                        help='GCS path to output CSV file')
-    parser.add_argument('--workflow-ids', metavar='WORKFLOWIDS', type=workflow_regex_validator, nargs='+',
-                        help='Workflow ids for performance comparison')
+    parser.add_argument('--force', action='store_true',
+                        help='Whether to force the comparer to run through certain warning conditions')
+    parser.add_argument('--name1', nargs=1, metavar='NAME_FOR_DIGEST_1', type=str,
+                        required=True, help='Name to use for the first digest')
+    parser.add_argument('--name2', nargs=1, metavar='NAME_FOR_DIGEST_2', type=str,
+                        required=True, help='Name to use for the second digest')
+    parser.add_argument('--digest1', nargs=1, metavar='PATH_TO_DIGEST_1', type=validate_path,
+                        required=True, help='First digest path to compare.')
+    parser.add_argument('--digest2', nargs=1, metavar='PATH_TO_DIGEST_2', type=validate_path,
+                        required=True, help='Second digest path to compare.')
+    parser.add_argument('--output-path', metavar='OUTPUT_PATH', type=validate_path, nargs=1,
+                        required=True, help='Path to output CSV file.')
+    parser.add_argument('--call-prefix-to-remove', metavar='CALL_PREFIX_TO_REMOVE', type=str, nargs='*',
+                        help='Call prefix to remove if present.')
 
     args = parser.parse_args()
     set_log_verbosity(args.verbose)
     quieten_chatty_imports()
     logger.info("Starting Comparer operation.")
 
-    credentials, project_id = google.auth.default()
-    storage_client = storage.Client(credentials = credentials)
-    input_gcs_bucket, input_gcs_path = args.digest_gcs_base_path[0]
+    _json_1, _json_2 = [json_from_path_string(p[0]) for p in [args.digest1, args.digest2]]
 
-    workflow_ids_and_jsons = read_digester_jsons_from_gcs(input_gcs_bucket, input_gcs_path, args.digester_version[0], args.workflow_ids, storage_client)
-    comparison_result_df = compare_jsons(workflow_ids_and_jsons)
-    result_csv_string = comparison_result_df.to_csv()
+    prefixes = [] if not args.call_prefix_to_remove else args.call_prefix_to_remove
 
-    output_gcs_bucket, output_gcs_path = args.output_gcs_file_path[0]
-    upload_blob(output_gcs_bucket, result_csv_string, output_gcs_path, storage_client, logger)
+    comparison_data = compare_jsons(_json_1, _json_2, args.name1[0], args.name2[0], prefixes, args.force)
+    out_path = args.output_path[0]
+    out = ComparisonPath.create(out_path)
+    if out.exists() and not args.force:
+        raise ValueError(f"Specified output file '{out_path}' already exists and --force not specified.")
+    out.write_text(csv_string_from_data(comparison_data))
```

Comment:
I thought we wanted to upload the comparison result to GCS?

Reply:
`out` is a `ComparisonPath`, so if the value of `out` is a GCS path then this will write to GCS.